openedx/features/content_tagging/models.py

```
@@ -0,0 +1,121 @@
+        max_length=3,
+        choices=RelType.choices,
+        default=RelType.OWNER,
+    )
+
+    class Meta:
+        indexes = [
+            models.Index(fields=["taxonomy", "rel_type"]),
+            models.Index(fields=["taxonomy", "rel_type", "org"]),
+        ]
+
+    @classmethod
+    def get_owners(
+        cls,
+        taxonomy: Taxonomy,
+    ) -> models.QuerySet:
+        """
+        Returns the org "owners" on the given taxonomy.
+
+        If the returned list is empty, then the returned taxonomy is enabled for all organizations.
+        """
+        return cls.objects.filter(
+            taxonomy=taxonomy,
+            rel_type=cls.RelType.OWNER,
+        )
+
+    @classmethod
+    def is_owner(cls, taxonomy: Taxonomy, org_short_name: str) -> bool:
+        """
+        Returns True if there's an "owner" relationship between the given taxonomy and an org with the given short name.
+
+        Or, if no "owner" relationships exist for the given taxonomy, then all orgs are enabled for that taxonomy.
+        """
+        org_owners = cls.get_owners(taxonomy)
+        if not org_owners.exists():
+            return True
+        org_owners = org_owners.filter(org__short_name=org_short_name)
+        return org_owners.exists()
+
+
+class ContentTag(ObjectTag):
+    """
+    ObjectTag that requires an opaque key as an object ID.
+    """
+
+    class Meta:
+        proxy = True
+
+    OPAQUE_KEY_TYPES = [
+        UsageKey,
+        CourseKey,
+        LibraryLocatorV2,
+        LibraryUsageLocatorV2,
+    ]
+
+    @property
+    def object_key(self) -> OpaqueKey:
+        """
+        Returns the object ID parsed as an Opaque Key, or None if invalid.
+        """
+        if self.object_id:
+            for OpaqueKeyClass in (UsageKey, CourseKey):
+                try:
+                    return OpaqueKeyClass.from_string(str(self.object_id))
+                except InvalidKeyError:
+                    pass
+        return None
```

Shouldn't this throw an exception rather than silently returning None?

---

Sure, can do it that way: https://github.com/openedx/edx-platform/pull/32661/files#diff-8a644422ddd022128d13b6ca7c8b672a33d048d1816a80a80491246bdd76f607R77-R88